modules/backend/classes/Controller.php

```diff
@@ -158,14 +158,6 @@ public function __construct()
          */
         $this->user = BackendAuth::getUser();
 
-        /*
-         * Media Manager widget is available on all back-end pages
-         */
-        if ($this->user && $this->user->hasAccess('media.*')) {
-            $manager = new MediaManager($this, 'ocmediamanager');
-            $manager->bindToController();
-        }
```

Comment:
@alxy @bennothommo doesn't this mean that the MediaManager widget is now bound to the controller once for every single instance of a field that uses it? That seems a bit excessive.

Reply:
Is there any way we could make a FormWidget trait for including the media manager in formwidgets so that it detects when the media manager widget has been already added to the controller and only adds it once if it hasn't already been added? That way we could also get rid of all the changes for dynamically changing the alias referenced.

Reply:
@LukeTowers correct, but I figure it's no different than the Repeater widget doing the same thing.

Reply:
It is a bit different actually. For FormWidgets, there should be a per-field instance of the FormWidget as each field could be different in its configuration. However, this is slightly different in that the Media Manager Widget (not formwidget) is a common tool that is used to access / manage the media library in October, and it just so happens that lots of different FormWidgets use this tool. Basically, there should be one "true" media manager, and however many instances of the FormWidgets that use it as necessary. Thoughts? @bennothommo

Reply:
@LukeTowers My thinking was that given each form widget that uses the Media Manager having the potential to affect operation of the Media Manager in some way (mainly from the `readOnly` and `disabled` properties), it might be safer for them to operate with different instances. All that being said, I haven't really studied the effects of the differences between how @alxy has set up and the normal operation beyond just testing that it works on a user level.

Reply:
But that would mean we need to initialize the "true" media manager "somewhere". Right now it is done in a central place for each and every backend request. I think its far more sensible to just initialize it when needed, even though that might mean we need to create multiple instances of the widget. (I don't see another nice solution to this problem.)

Reply:
@bennothommo I believe the readOnly property is passed to the media manager via JS so that shouldn't be a problem.

@alxy the idea is that there would be a common trait that every widget would call `$this->bindMediaManager()` and it would check the controller instance to make sure the media manager is only ever bound once. To be clear, my main concerns with the approach taken in this PR is the passing around of `mediaManagerAlias` everywhere and the repeating of that code; then as a secondary concern the added performance overhead of intializing it multiple times (although I haven't actually run any performance testing so I'm not going to be too pushy on that as changes for the sake of performance without actual numbers to go by are more harmful than helpful most of the time).